### PR TITLE
Add Azure Queue Storage support

### PR DESCRIPTION
**Pull request by subhra231**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/.editorconfig [hunk 1]
```diff
@@ -0,0 +1,199 @@
+# top-most EditorConfig file
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/.editorconfig:1]: Please remove;  we'll want to conform to the standards from the repository root .editorconfig

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/Azure.CoreWCF.AzureQueueStorage.sln [hunk 2]
```diff
@@ -0,0 +1,81 @@
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/Azure.CoreWCF.AzureQueueStorage.sln:1]: Unless we're planning on strictly adopting the API patterns in the [Azure SDK Guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html), these packages cannot use the `Azure.*` prefix.   We'll need to move them over to `Microsoft.Azure.*`.

If we are planning to adhere to the SDK guidelines, we'll need to start a conversation with the archboard - particularly @KrzysztofCwalina and @tg-msft - around the namespace.   `CoreWCF` is not currently an approved service area.

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/README.MD [hunk 3]
```diff
@@ -0,0 +1,9 @@
+Using the CoreWCF Postman Samples Collection.json within Postman
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/README.MD:1]: We'll need to update this to follow the README conventions described in [the guidelines](https://azure.github.io/azure-sdk/general_documentation.html).

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/NuGet.config [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/NuGet.config:3]: This cannot be included when the PR is merged.   The Azure SDK repository only permits package references from NuGet or the Azure SDK development feed.

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/Azure.CoreWCF.AzureQueueStorage.csproj [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <AssemblyName>Azure.CoreWCF.AzureQueueStorage</AssemblyName>
+    <PackageId>Azure.CoreWCF.AzureQueueStorage</PackageId>
+    <RootNamespace />
+    <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Storage.Queues" />
+    <PackageReference Include="NuGet.Commands" VersionOverride="6.5.0" />
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/Azure.CoreWCF.AzureQueueStorage.csproj:13]: These dependencies will need to be approved by the architects and added to the [central package versions](https://github.com/Azure/azure-sdk-for-net/blob/main/eng/Packages.Data.props).   We generally do not want to include local `VersionOverrides` unless there's a temporary exceptional case.

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageBinding.cs [hunk 6]
```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Net;
+using Azure.Storage.Queues;
+using CoreWCF.Channels;
+
+namespace Azure.Storage.CoreWCF.Channels
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageBinding.cs:8]: We'll need to review namesapces here.   I do not believe that we'd want these to start with `Azure.Storage`.


//cc: @KrzysztofCwalina, @tg-msft

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageBinding.cs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+{
+    /// <summary>
+    /// The class storing binding information
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageBinding.cs:11]: The doc comments for public members will be rendered in MS Docs.   I believe that we'll need to spend some time tuning these as customer-facing documentation.

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageQueueNameConverter.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    /// Class containing methods to convert Queue Name.
+    /// </summary>
+    public class AzureQueueStorageQueueNameConverter
+    {
+        /// <summary>
+        /// Method to fetch Queue Endpoint Url
+        /// </summary>
+        public static string GetEndpointUrl(string accountName, string queueName)
+        {
+            return $"net.aqs://{accountName}.queue.core.windows.net/{queueName}";
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/src/CoreWCF/Channels/AzureQueueStorageQueueNameConverter.cs:20]: I do not believe this works across Azure clouds and with Azure stack, as written.

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/tests/Azure.CoreWCF.AzureQueueStorage.Tests.csproj [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </Content>
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Storage.Blobs" />
+    <PackageReference Include="Microsoft.AspNetCore" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.UserSecrets" VersionOverride="3.1.13" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" />
+    <PackageReference Include="xunit" />
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/tests/Azure.CoreWCF.AzureQueueStorage.Tests.csproj:29]: The Azure SDK repository is standardized on NUnit, outside of some exceptions for deprecated legacy packages.    We'll want to convert over from xunit.

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/tests/IntegrationTests.cs [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+using Xunit.Abstractions;
+
+namespace CoreWCF.AzureQueueStorage.Tests
+{
+    public class IntegrationTests
+    {
+        private readonly ITestOutputHelper _output = null;
+        public const string QueueName = "AzureQueue";
+
+        [Fact(Skip = "Need AzureQueue")]
```
Comment (jsquire) [inline sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/tests/IntegrationTests.cs:17]: We'll want to convert this over to the Azure SDK test infrastructure for live test support.

**Review by jsquire — COMMENTED** [review]
I realize this is still in draft, but I left some feedback around tweaks that we'll need to make to integrate into the Azure SDK repository and some conversations that we'll need to have with the architects around naming.

**Comment by jsquire** [comment]
Can we please shift the source path to `$/sdk/wcf` to better conform to the patterns used by the repository?

**Comment by jsquire** [comment]
//fyi:  @KrzysztofCwalina, @tg-msft, @schaabs

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/.editorconfig (hunk 1 above)
Comment (subhra231) [reply sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/.editorconfig]: Makes sense. Removed it.

File: sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/NuGet.config (hunk 4 above)
Comment (subhra231) [reply sdk/CoreWCF/Azure.CoreWCF.AzureQueueStorage/NuGet.config]: This was actually the remnant from our previous code. We no longer need the config file. so removed it

**Comment by subhra231** [comment]
> Can we please shift the source path to `$/sdk/wcf` to better conform to the patterns used by the repository?

Sure, Have updated the path to sdk/wcf

**Pull request closed without merging**